apartmenttherapy/order-service
# Elaine/pay business logic

#### What does this PR do & why?
Adding some buyer protection when they go to pay. We want to make sure the listing they're paying for is still open and published.

#### How can this be manually tested?
- Tests should run.
- Open and accept an order. Do any of the following to the listing:
  - Close it
  - Revert it to draft
  - Delete it
Then try to pay for the listing. You should get an error.

#### Any questions or concerns about this change?
*(Any significant changes you're introducing? What side effects might this have? What other parts of the app could this affect?)*


#### Deployment Notes
*(e.g. Does this require a migration or special infrastructure change? Clearing the cache? Third party service?)*


#### Link to story card and other documentation
https://app.asana.com/0/694906336753887/1114280868528483

#### GIF


#### Have you remembered to...?
